dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -117,7 +128,21 @@ private void parseRecursive(Reader source, Listener listener) throws IOException
                 break;
             }
             buffer.add((char) readCharacter);
-            if (getStringRepresentation(buffer).endsWith("----")) {
+
+            List<DokuWikiPlugin> componentList = null;
+            if (componentManagerProvider != null) {
+                try {
+                    componentList = componentManagerProvider.get().getInstanceList(DokuWikiPlugin.class);
```

Comment:
You should do that at the beginning of the method before the while, right now you are doing it for each character which does not make much sense.